### PR TITLE
enhancement: properly produce raw format data to Kafka external streams

### LINKED ISSUE #435: Write to Kafka external stream with raw data format should only send the raw data to Kafka
**Describe what enhancement you'd like to have**

Currently, when `data_format` is not speicified in `CREATE EXTERNAL STREAM` with `type='kafka'` and there is only one string value column, it uses the Raw data format.

When reading from such a stream, the behavior is correct, it uses the message payload as-is as the column value. However, when writing to such a stream, it writes a JSON object, something like `{"data": "<the-raw-value>"}`. This is not the correct raw format behavior, this is exactly the same as using `JSONEachRow` format. The correct behavior for raw writing is to just write the string value as the Kafka message payload without adding any other thing to it. Simply put with the following SQL:

```sql
INSERT INTO my_raw_kafka_external_stream SELECT FROM my_raw_kafka_external_stream
```

should keep producing the unique message again and again to the Kafka topic.

### PR DESCRIPTION
**Pull request by zliang-min**
PR checklist:
- Did you run ClangFormat ?
- Did you separate headers to a different section in existing community code base ?
- Did you surround `proton: starts/ends` for new code in existing community code base ?

Please write user-readable short description of the changes:

closes #435 

Please check the issue description for the desired behavior for raw writting.

### CODE REVIEW

File: src/Storages/ExternalStream/Kafka/Kafka.cpp [hunk 1]
```diff
@@ -180,24 +180,19 @@ void Kafka::calculateDataFormat(const IStorage * storage)
...
-            ErrorCodes::INVALID_SETTING_VALUE,
-            "`data_format` settings is empty but the Kafka external stream definition has multiple columns. Proton doesn't know how to "
-            "parse Kafka messages without a data format.");
-
-    auto type = column_names_and_types.begin()->type;
-    if (type->getTypeId() == TypeIndex::String || type->getTypeId() == TypeIndex::FixedString)
+    if (column_names_and_types.size() == 1)
     {
-        /// no-op
+        auto type = column_names_and_types.begin()->type;
```
Comment (chenziliang) [inline src/Storages/ExternalStream/Kafka/Kafka.cpp]: auto type = [column_names_and_types.begin()->type->getTypeId();

File: src/Storages/ExternalStream/Kafka/Kafka.cpp [hunk 2]
```diff
@@ -180,24 +180,19 @@ void Kafka::calculateDataFormat(const IStorage * storage)
...
+        if (type->getTypeId() == TypeIndex::String || type->getTypeId() == TypeIndex::FixedString)
+        {
+            data_format = "RawBLOB";
```
Comment (chenziliang) [inline src/Storages/ExternalStream/Kafka/Kafka.cpp:193]: Is it `RawBlob` ?

**Comment by chenziliang** [comment]
Other changes look good

File: src/Storages/ExternalStream/Kafka/Kafka.cpp (hunk 2 above)
Comment (zliang-min) [reply src/Storages/ExternalStream/Kafka/Kafka.cpp:193]: No, it's `RawBLOB`.

Comment (chenziliang) [reply src/Storages/ExternalStream/Kafka/Kafka.cpp:193]: It is our new format or existing one since it doesn't seem follow our name convention

**Comment by jovezhong** [comment]
(Jove Github Bot) added it to the current sprint.

**Comment by jovezhong** [comment]
(Jove Github Bot) moved this ticket out of the GitHub project(up to 1200 tickets for one project).

**Pull request merged**